design/client.md

```diff
@@ -0,0 +1,23 @@
+# Client
+
+Since containerd uses [gRPC](https://grpc.io/), it should be easy to implement 3rd-party containerd client in almost any language.
+
+<!--- TODO: more information about how containerd is a client-heavy design --->
+
+However, 3rd-party client would need to reimplement some operations, which are not implemented on the daemon side:
+
+- Pull & Push images from remote registries, such as OCI images via Docker Registry
+- Import & Export local image archives
+- Generate container runtime configuration, such as OCI Runtime Spec
+- Chown files under the container rootfs for remapping UID / GID
+
+## Label conventions
+
+Client implementations SHOULD follow this labeling convention so as to promote compatibility across different implementations.
+
+Key (omitted prefix: `containerd.io/`)  | Object                                           | Value
+----------------------------------------|--------------------------------------------------|--------------------------------------------------
+`uncompressed`                          | Content (compressed layer blobs), Snapshot       | Digest string of uncompressed layer blob.
+`gc.root`                               | Content, Snapshot                                | RFC3339 UTC timestamp string. Objects with this label are treated as the roots for GC.
+`gc.ref.content`                        | Content                                          | Digest string of children blobs. This label represents Content->Content edges for GC.
```

Comment:
Also allow `gc.ref.content.${something}` to allow adding multiple links. It would be inaccurate to say "Digest string of children blobs." since each label may only refer to a single digest. To refer to multiple children multiple label keys must be used using a client define string after `gc.ref.content.`.